File: FirmaTransportSockets/Services/IService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;
using Persistence;

namespace Services
{
    public interface IService
    {
        Oficiu Login(string nume, string parola,IFTObserver client);
        void Logout(string nume, IFTObserver client);
        IEnumerable<Cursa> findAllCursa();
        IEnumerable<CursaDTO> findAllCursaDTO();
        IEnumerable<Cursa> findCursaDTOByDestinatie(string destinatie);
        IEnumerable<RezervareDTO> findRezervareDTO(Cursa c);
        Rezervare saveRezervare(string numeClient, int nrLocuri, int idCursa);
    }
}

File: FirmaTransportSockets/Services/src/main/java/services/IService.java
package services;

import java.util.List;
import domain.Cursa;
import domain.Oficiu;
import domain.Rezervare;
import domain.RezervareDTO;

public interface IService {
    Oficiu login(String nume, String parola,IObserver client) throws FTException;
    List<Cursa> findAllCurse() throws FTException;
    List<Rezervare> findAllRezervari() throws FTException;
    List<RezervareDTO> findRezervareDTO(String id1) throws FTException;
    List<Cursa> findByDestinatie(String destinatie) throws FTException;
    Rezervare saveRezervare(Rezervare rezervare) throws  FTException;
    void logout(String nume, IObserver client) throws  FTException;


}
